### DISCUSSION TITLE
Issue with Tailwind CSS output class order.

### DISCUSSION THREAD

**Discussion by pieter-berkel** [tailwindlabs/tailwindcss#13059] [Help]
After converting my polyrepo to a monorepo, I've noticed a discrepancy in the Tailwind CSS output. Specifically, I'm encountering a situation where non-media queries are overwriting media queries. Below, I've included a snippet of the output file to illustrate the order of classes:
```
.flex-row {
  flex-direction: row;
}
.flex-col {
  flex-direction: column;
}

@media (min-width: 768px) {
  .md\:flex-row {
    flex-direction: row;
  }
}

.flex-col {
  flex-direction: column
}
```

The issue arises when I use the class flex flex-col md:flex-row inside a div. It seems that the flex-col class always takes precedence, and flex-row never gets applied because flex-col appears later in the CSS file.

Could someone shed some light on why this behavior is occurring and suggest possible solutions? Any help would be greatly appreciated!

tailwind.config.ts
```
import type { Config } from "tailwindcss";

const config: Config = {
  darkMode: ["class"],
  content: ["./src/**/*.{ts,tsx}"],
  plugins: [require("tailwindcss-animate")],
};

export default config;
```

globals.css
```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  /* color codes removed for clarity */
}

@layer base {
  * {
    @apply border-border;
  }

  html {
    @apply h-full;
  }

  body {
    @apply min-h-full bg-background font-sans text-foreground antialiased;
    font-feature-settings:
      "rlig" 1,
      "calt" 1;
    overscroll-behavior: none;
  }

  #__next {
    @apply min-h-full;
  }
}

@media (max-width: 640px) {
  .container {
    @apply px-4;
  }
}
```

**Comment by alex-krasikau** [comment]
Hey @pieter-berkel!

Can you please try stopping your project, cleaning up the cache, trying the production build, and checking if you still observe the issue? When running Tailwind CSS in watch mode, sometimes it messes up the class declaration order.
